### DISCUSSION TITLE
Browser extensions for adding feeds

### DISCUSSION THREAD

**Discussion by JayBrown** [ViennaRSS/vienna-rss#1581] [Ideas]
Coming from **[this issue](https://github.com/ViennaRSS/vienna-rss/issues/1234)**.

It would be great if Vienna came with installable browser addons (in this case for Firefox) that give you a drop-down menu with all RSS feed on a webpage, and when you click on one of them, a new subscription will automatically be added to Vienna.

Additional nice preference settings would be an option to add the Vienna RSS button into the URL bar, and support for dark mode (toggle on/off), i.e. a light and dark icon for the extension.

In case of Safari extensions, they will automatically register with the Safari app from within the Vienna app bundle. Extensions for Firefox (and Chrome, too, I assume) must be installed manually, so there would need to be a section in Vienna preferences to install the addons.

Possible?

**Comment by Eitot** [comment]
It is certainly possible. Vienna supports importing of feed URLs via AppleScript.

A Safari app extension is already on the list:  [Cocoa modernization](https://github.com/ViennaRSS/vienna-rss/projects/5). Alas, it isn’t a high priority.

**Comment by barijaona** [comment]
This is not per se a Vienna problem : asking Firefox or Chrome to open a `feed://` URI do the trick.

I don't know why developers of browser extensions don't do this. Yes, it is not a largely recognized standard, but [it is documented](https://www.iana.org/assignments/uri-schemes/prov/feed) and should run cross platform.
